org.eclipse.m2e.core.ui/src/org/eclipse/m2e/core/ui/internal/project/MavenUpdateConfigurationChangeListener.java

```diff
@@ -52,10 +53,12 @@ public void resourceChanged(IResourceChangeEvent event) {
       LOG.error("An error occurred while checking for out-of-date configuration markers", e);
       return;
     }
+    outOfDateProjects = outOfDateProjects.stream() //
+        .filter(ResolverConfigurationIO::isAutomaticallyUpdateConfiguration).toList();
     updateProjectConfiguration(outOfDateProjects);
   }
 
-  private boolean isDisabled() {
+  public static boolean isAutoConfigurationUpdateDisabled() {
```

Comment:
I would add an IProject parameter here then it could:

1. Check if there is a project config and if yes choose this one unless it is configured to use the workspace default
2. fallback to the global default

```suggestion
  public static boolean isAutoConfigurationUpdate(IProject) {
```